src/test/java/io/conduit/SparkDestinationStreamIT.java

```diff
@@ -168,31 +159,96 @@ void testInsertStructured() {
         verify(observerMock, never()).onError(any());
 
         var foundRecords = readIcebergRecords();
-        assertEquals(3, foundRecords.size());
-        assertOk(foundRecords.get(2), eventTime);
+        assertEquals(1, foundRecords.size());
+        assertOk(foundRecords.get(0), eventTime);
     }
 
     @Test
-    void testDelete() {
+    void testDeleteStructuredKey() {
+        testDeleteWithKey(
+            Data.newBuilder()
+                .setStructuredData(Struct.newBuilder()
+                    .putFields("integer_field", Value.newBuilder()
+                        .setNumberValue(12)
+                        .build())
+                    .build())
+                .build()
+        );
+    }
+
+    @Test
+    void testDeleteRawDataKey() {
+        testDeleteWithKey(
+            Data.newBuilder()
+                .setRawData(ByteString.copyFromUtf8("""
+                    {
+                      "integer_field": 12
+                    }
+                    """))
+                .build()
+        );
+    }
+
+    @Test
+    void testDeleteWithMaliciousKey() {
         var observerMock = Mockito.mock(StreamObserver.class);
-        SparkDestinationStream stream = new SparkDestinationStream(observerMock, spark, config.getCatalogName() + "." + config.getNamespace() + "." + config.getTableName());
+        insertTestRecord("testDelete_record", 12);
+        insertTestRecord("testDelete_record", 34);
+
+        SparkDestinationStream stream = new SparkDestinationStream(observerMock, spark, config.fullTableName());
         stream.onNext(
             Request.newBuilder()
                 .setRecord(Record.newBuilder()
                     .setKey(
                         Data.newBuilder()
-                            .setStructuredData(Struct.newBuilder()
-                                .putFields("integer_field", Value.newBuilder()
-                                    .setStringValue("12")
-                                    .build())
-                                .build())
+                            .setRawData(ByteString.copyFromUtf8("""
+                                {
+                                  "integer_field": "105 OR 1=1"
+                                }
+                                """))
+                            .build()
                     ).setOperation(Operation.OPERATION_DELETE)
                     .build()
                 ).build()
         );
+        verify(observerMock).onNext(any());
+        verify(observerMock, never()).onError(any());
+
+        var foundRecords = readIcebergRecords();
+        assertEquals(2, foundRecords.size());
+    }
+
+    private void testDeleteWithKey(Data key) {
+        var observerMock = Mockito.mock(StreamObserver.class);
+        insertTestRecord("testDelete_record", 12);
+        insertTestRecord("testDelete_record", 34);
```

Comment:
is there a specific reason why we're inserting two records and not one for the delete tests? If not, I think we could go for one, just saving time